### ISSUE TITLE
$elemMatch is not reset correctly

### ISSUE THREAD

**Issue by stalniy** [crcn/sift.js#212]
## Code

```js
const sift = require('sift');


const test = sift({
  groups: {
    $elemMatch: { id: 1, status: 'review' }
  }
});

const items = [
  { groups: [{ id: 1, status: 'review' }] },
  { groups: [{ id: 1, status: 'done' }] },
];

console.log(0, 'expects true', test(items[0]));
console.log(1, 'expects false', test(items[1])); // returns true!!
```

## Explanation

When `$elemMatch` evaluates to `true` it will never evaluate to `false` anymore!!! This is because [`$ElemMatch`  class](https://github.com/crcn/sift.js/blob/master/src/operations.ts#L44) incorrectly overrides `reset` method of `BaseOperation`.

## Possible fix

So, `reset` method of `$ElemMatch` should be this:

```js
  reset() {
    super.reset(); // <--- added this line
    this._queryOperation.reset();
  }
```

## Importance

It's a **critical issue** for casl v4, so if you are OK with the suggested fix, I'll send a PR. Related issue https://github.com/stalniy/casl/issues/392

**Comment by stalniy** [comment]
@crcn ping

**Comment by crcn** [comment]
Fixed! Should be up on NPM as `13.3.4`.
